src/posts/interning-at-khan-academy.md

```diff
@@ -0,0 +1,53 @@
+
+As part of the XOM initiative, the team had designed a math keypad that could be used to type in expression responses to mathematics problems in a mobile-friendly way. I started by fixing some bugs to get familiar with the math keypad codebase, such as disabling gestures on the static sidebar, a fixed bar on the side of the keypad with navigation buttons. Once I had a good idea of how the keypad worked, I was able to start working on adding new features our designers had crafted. I worked on things such as using square icons for representing empty blocks, and learned in the process how important it is to make these small changes to create a great experience for students.
+
+At Khan Academy, we use React.js, something I had worked with before, but only in the context of [Scala.js](https://www.scala-js.org) applications. As part of making changes to the math keypad, I had to learn how to write idiomatic ES6 code. Working on Khan Academy gave me a great opportunity to use React with ES6 in a production environment, and left me with many ideas on how React.js could be used more seamlessly with Scala.js. In addition to learning React and ES6, I got a chance to work with the plethora of open-source libraries at Khan Academy such as [Aphrodite](https://github.com/khan/aphrodite) and [KaTeX](https://github.com/khan/katex).
+
+![changes to the math keypad](/images/interning-at-khan-academy/image_0.png)
+
+*an early screenshot of changes to the math keypad*
+
+# Part 2: iframes, iframes, iframes
+
+Now that I was used to the flow of picking a task, creating a code review, and deploying my changes, I started on my first project -- improving our content editor to be mobile first. This involved pulling together specs for what types of devices students use Khan Academy on, rendering content with the new styles the rest of the team was working on, isolating the rendered content to prevent styles leaking from the rest of the editor, and implementing touch emulation for interacting with the mobile previews.
+
+To implement the isolation of content, I decided to render previews inside of iframes. In addition to providing style isolation, iframes also made it possible to properly render components that are rendered directly to the body, such as the math keypad I worked on earlier. To transfer data between the parent editor and the child iframe, the parent first stores a JS object representation of the exercise content in a global variable, because the exercise content contains callback functions which prevent it from being serialized for window messaging. The parent then sends a message ([http://ejohn.org/blog/cross-window-messaging/](http://ejohn.org/blog/cross-window-messaging/)) to the iframe, notifying it of the updated content.
```

Comment:
In this paragraph and the one below, I think the storytelling pattern of "There was this problem, wow, interesting! Here's how we solved it." would help. Technical audiences love that: they get to chew on the problem themselves and go "oh, huh" before seeing your solution :) It also raises the stakes: instead of describing a thing you did, you're describing a potential deal-breaker of a problem that _you_ overcame :D